e2e/test/src/r13n_brick/r13n_brick_test.dart

```diff
@@ -0,0 +1,237 @@
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:mason/mason.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+
+/// Exit code indicating a command completed successfully.
+///
+/// [Source](https://www.freebsd.org/cgi/man.cgi?query=sysexits).
+const _sucessfulExitCode = 0;
+
+/// Objectives:
+///
+/// * Generate AppRegionalizations from arb files (`mason make r13n`)
+/// * Ensure the code is formatted (`dart format .`)
+/// * Ensure the code has no warnings/errors (`dart analyze .`)
+/// * Ensure generated AppRegionalizations match the expected output (fixtures/gen/)
+void main() {
+  test(
+    'r13n brick generates successfully',
+    timeout: const Timeout(Duration(minutes: 5)),
+    () async {
+      final tempDirectory = Directory.systemTemp.createTempSync();
+
+      final arbPath = path.join(tempDirectory.path, 'lib', 'r13n', 'arb');
+      Directory(arbPath).createSync(recursive: true);
+
+      final arbUsPath = path.join(arbPath, 'app_us.arb');
+      const arbUsFileContent = '''
+{
+    "@@region": "us",
+    "supportEmail": "us@verygood.ventures"
+}
+''';
+
+      File(arbUsPath)
+        ..writeAsStringSync(arbUsFileContent)
+        ..createSync();
+
+      final arbGbPath = path.join(arbPath, 'app_gb.arb');
+      const arbGbFileContent = '''
+{
+    "@@region": "gb",
+    "supportEmail": "gb@verygood.ventures"
+}
+''';
+      File(arbGbPath)
+        ..writeAsStringSync(arbGbFileContent)
+        ..createSync();
+
+      final arbDirectory = path.relative(arbPath, from: tempDirectory.path);
+      final templateArbFile = path.basename(arbUsPath);
+      final r13nYamlPath = path.join(tempDirectory.path, 'r13n.yaml');
+      final r13nYamlFileContent = '''
+arb-dir: $arbDirectory
+template-arb-file: $templateArbFile
+''';
+      File(r13nYamlPath)
+        ..writeAsStringSync(r13nYamlFileContent)
+        ..createSync();
+
+      final pubspecYalPath = path.join(tempDirectory.path, 'pubspec.yaml');
+      const pubsecYamlFileContent = '''
+name: r13n_e2e_test
+description: An r13n e2e test.
+publish_to: "none"
+
+environment:
+  sdk: ">=2.18.0 <3.0.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  r13n: 0.1.0-dev.2
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  very_good_analysis: ^3.1.0
+''';
+      File(pubspecYalPath)
+        ..writeAsStringSync(pubsecYamlFileContent)
+        ..createSync();
+
+      final flutterPubGetResult = await Process.run(
+        'flutter',
+        ['pub', 'get'],
+        runInShell: true,
+      );
```

Comment:
Any ideas on why awaiting the `Process.run` doesn't wait for the command to finish? The `pubspec.lock` seems to be generated after the completion of the `Future`.

(cc: @renancaraujo , @wolfenrain )